packages/uicore/src/components/ExpressionInput/ExpressionInput.tsx

```diff
@@ -181,14 +183,44 @@ export function ExpressionInput(props: ExpressionInputProps): React.ReactElement
         window.requestAnimationFrame(() => {
           if (inputRef.current) {
             // position is sum of firstHalf.length + 2 (for '<+') + item.length + 1 (for '>')
-            const position = firstHalf.length + 2 + item.length + 1
+            if (newExpressionComponent) {
+              const position = firstHalf.length + 2 + item.length + 2
```

Comment:
move the comment at line 185 to the `else` block and add a new comment for this `position`